app/src/org/commcare/activities/FormEntryActivity.java

```diff
@@ -257,8 +257,9 @@ public void onReceive(Context context, Intent intent) {
     @Override
     protected void onSaveInstanceState(Bundle outState) {
         super.onSaveInstanceState(outState);
-        instanceState.saveState(outState);
-
+        if (instanceState != null) {
```

Comment:
What scenarios lead to `instanceState` being null ?

Reply:
I honestly have no idea why it would ever be null here. I thought that this might be a result of using IntentTestRule [here](https://github.com/dimagi/commcare-android/pull/2231/files#diff-14085b87043ea3b3b0ad6a0f2be4f793R59) since the crash started happening after adding that only and it's needed to stub intents. 

But then I also saw some crashes related to this in [crashlytics](https://console.firebase.google.com/u/0/project/commcare-a57e4/crashlytics/app/android:org.commcare.dalvik/issues/d3f3ff13c96e62cb358bc94c1fc8fc99?time=last-seven-days&sessionId=5EC12E04004C00015A3462220FB79688_DNE_0_v2) as well.

Reply:
Would be good to investigate this if you are encountering it in your test,  since doing this null check will only result in supressing errors we should not and can result in user believing everything is alright when it is not.

Reply:
will remove this change from the PR before merging.